angular/projects/spark-angular/src/lib/components/sprk-tabs/sprk-tabs.component.spec.ts

```diff
@@ -0,0 +1,474 @@
+      <div sprkTabsPanel>Tab3 Content</div>
+    </sprk-tabs>
+  `,
+})
+class TestComponent {}
+
+@Component({
+  selector: 'sprk-dynamic-test',
+  template: `
+    <div>
+      <button class="changeButton" (click)="changeMe()">Change Content</button>
+      <sprk-tabs idString="tabs-1">
+        <button
+          *ngFor="let item of list; let i = index"
+          sprkTabsButton
+          analyticsString="Tab: {{ i }}"
+        >
+          {{ item }}
+        </button>
+        <div sprkTabsPanel *ngFor="let item of list; let i = index">
+          <p>{{ item }}</p>
+        </div>
+      </sprk-tabs>
+    </div>
+  `,
+})
+class DynamicComponent {
+  public list = ['Tab 33', 'Tab 2'];
+  public newList = ['Tab 3'];
+  changeMe(): void {
+    this.list = this.newList;
+  }
+}
+
+describe('SprkTabsComponent', () => {
+  let component: SprkTabsComponent;
+  let fixture: ComponentFixture<SprkTabsComponent>;
+  let element: HTMLElement;
+
+  let dynamicTestComponent;
+  let dynamicTestFixture;
+
+  let testComponent: TestComponent;
+  let testFixture: ComponentFixture<TestComponent>;
+  let testElement;
+  let testTab1;
+  let testTab2;
+  let testTab3;
+  let testPanel1;
+  let testPanel2;
+  let testPanel3;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        SprkTabsComponent,
+        TestComponent,
+        DynamicComponent,
+        SprkTabsPanelDirective,
+        SprkTabsButtonDirective,
+      ],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SprkTabsComponent);
+    component = fixture.componentInstance;
+
+    testFixture = TestBed.createComponent(TestComponent);
+    testComponent = testFixture.componentInstance;
+    testFixture.detectChanges();
+
+    element = fixture.nativeElement.querySelector('div');
+    testElement = testFixture.nativeElement.querySelector('sprk-tabs');
+
+    testTab1 = testElement.querySelectorAll('.sprk-c-Tabs__button')[0];
+    testTab2 = testElement.querySelectorAll('.sprk-c-Tabs__button')[1];
+    testTab3 = testElement.querySelectorAll('.sprk-c-Tabs__button')[2];
+    testPanel1 = testElement.querySelectorAll('.sprk-c-Tabs__content')[0];
+    testPanel2 = testElement.querySelectorAll('.sprk-c-Tabs__content')[1];
+    testPanel3 = testElement.querySelectorAll('.sprk-c-Tabs__content')[2];
+
+    // Dynamic Test Component
+    dynamicTestFixture = TestBed.createComponent(DynamicComponent);
+    dynamicTestComponent = dynamicTestFixture.componentInstance;
+  });
+
+  it('should create itself', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should add classes if additionalClasses has a value', () => {
+    component.additionalClasses = 'sprk-u-man';
+    fixture.detectChanges();
+    expect(element.classList.toString()).toEqual('sprk-c-Tabs sprk-u-man');
+  });
+
+  it('should add tabstrip additional classes if tablistAdditionalClasses has a value', () => {
```

Comment:
@bobwise , Can you update the react Tabs issue to account for this new additional classes addition. We want to make sure we add it there as well. Comment back here when complete. Thank you. #3297

Reply:
Added. This AC is also already in the deprecation strategy document that we should be following for these stories.